### PR TITLE
 JAMES-2430 Write an integration test for DLP matcher 

### PR DESCRIPTION
**Pull request by chibenwa**
Review https://github.com/linagora/james-project/commit/01e2d8aa564696f12ff5fc9e108c523227db85b0 only

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[01e2d8aa564696f12ff5fc9e108c523227db85b0] docker: Error response from daemon: No such container: keys.
```

test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/DlpIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.inject.util.Modules;
+import com.jayway.restassured.specification.RequestSpecification;
+
+public class DlpIntegrationTest {
+    private static final String OTHER_DOMAIN = "other.org";
+    private static final String RECIPIENT3 = "user4@" + OTHER_DOMAIN;
```
Comment (rouazana) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/DlpIntegrationTest.java]: recipient3 is user4? :)

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/DlpIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        MailRepositoryUrl repositoryUrl = MailRepositoryUrl.from(REPOSITORY_PREFIX + DEFAULT_DOMAIN);
+        awaitAtMostOneMinute.until(() ->
+            given()
+                .spec(specification)
+                .get("/mailRepositories/" + repositoryUrl.urlEncoded() + "/mails")
+                .jsonPath()
+                .getList(".")
+                .size() == 1);
+    }
+}
```
Comment (rouazana) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/DlpIntegrationTest.java]: you should test repository creation options

**Comment by chibenwa** [comment]
test this please

**Comment by aduprat** [comment]
```
[ERROR] /james-project/server/data/data-cassandra/src/main/java/org/apache/james/dlp/eventsourcing/cassandra/DLPConfigurationItemDTO.java:[122,28] incompatible types: boolean cannot be converted to java.util.Optional<java.lang.Boolean>
```

**Comment by aduprat** [comment]
Rebased in order to fix compilation error (base on #1486)

**Comment by rouazana** [comment]
```
[4ac72f1d2c14cce6793da87351852181c0036c64]   DLPConfigurationStoreContract.storeShouldOverwritePreviouslyStoredEntries:100 
[4ac72f1d2c14cce6793da87351852181c0036c64] Expecting:
[4ac72f1d2c14cce6793da87351852181c0036c64]   <[DLPConfigurationItem{id=Id{value=id1}, explanation=Optional[explanation], regexp=regex, targets=Targets{targets=[Sender]}},
[4ac72f1d2c14cce6793da87351852181c0036c64]     DLPConfigurationItem{id=Id{value=id2}, explanation=Optional[explanation2], regexp=regex2, targets=Targets{targets=[Sender, Recipient]}},
[4ac72f1d2c14cce6793da87351852181c0036c64]     DLPConfigurationItem{id=Id{value=id1}, explanation=Optional[explanation], regexp=regex, targets=Targets{targets=[Sender]}}]>
[4ac72f1d2c14cce6793da87351852181c0036c64] to contain only:
[4ac72f1d2c14cce6793da87351852181c0036c64]   <[DLPConfigurationItem{id=Id{value=id1}, explanation=Optional[explanation], regexp=regex, targets=Targets{targets=[Sender]}}]>
[4ac72f1d2c14cce6793da87351852181c0036c64] elements not found:
[4ac72f1d2c14cce6793da87351852181c0036c64]   <[DLPConfigurationItem{id=Id{value=id1}, explanation=Optional[explanation], regexp=regex, targets=Targets{targets=[Sender]}}]>
[4ac72f1d2c14cce6793da87351852181c0036c64] and elements not expected:
[4ac72f1d2c14cce6793da87351852181c0036c64]   <[DLPConfigurationItem{id=Id{value=id1}, explanation=Optional[explanation], regexp=regex, targets=Targets{targets=[Sender]}},
[4ac72f1d2c14cce6793da87351852181c0036c64]     DLPConfigurationItem{id=Id{value=id2}, explanation=Optional[explanation2], regexp=regex2, targets=Targets{targets=[Sender, Recipient]}},
[4ac72f1d2c14cce6793da87351852181c0036c64]     DLPConfigurationItem{id=Id{value=id1}, explanation=Optional[explanation], regexp=regex, targets=Targets{targets=[Sender]}}]>
```

**Comment by aduprat** [comment]
Rebased again

**Comment by chibenwa** [comment]
DLP cassandra store

**Comment by chibenwa** [comment]
```
[afa3c751354dac5adf85af284f28bc4f1c9d71b3] Tests in error: 
[afa3c751354dac5adf85af284f28bc4f1c9d71b3]   DlpIntegrationTest.dlpShouldStoreMailWhenNotAllowedButRepositoryExists:260 » ConditionTimeout
[afa3c751354dac5adf85af284f28bc4f1c9d71b3]   DlpIntegrationTest.lambda$dlpShouldStoreMatchingEmails$3:148 » ClassCast java....
```

Will investigate

**Comment by chibenwa** [comment]
:trollface:  someone removed WebAdmon DLPConfigurationRoutes commits from https://github.com/linagora/james-project/pull/1485 while rebasing!

And of course it do work a bit less good!

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**